### ISSUE TITLE
RouteStatusGatewayReference.Controller should have an `omitempty` JSON tag

### ISSUE THREAD

**Issue by skriss** [kubernetes-sigs/gateway-api#669]
**What happened**:
Contour code that updates HTTPRoute status produces errors like the following: 
```
time="2021-05-05T17:02:01Z" level=error msg="unable to update status" context=StatusUpdateHandler error="HTTPRoute.networking.x-k8s.io \"http-filter-1\" is invalid: status.gateways.gatewayRef.controller: Invalid value: \"null\": status.gateways.gatewayRef.controller in body must be of type string: \"null\"" name=http-filter-1 namespace=gateway-005-request-header-modifier-rule
```

**What you expected to happen**:
Since the `Controller` field is optional, it should be omitted from JSON serialization if not specified.

**How to reproduce it (as minimally and precisely as possible)**:
I can provide more detail on how to run the Contour E2E's locally to observe this error if that's actually useful for anyone.

**Anything else we need to know?**:
Happy to submit a PR, just need confirmation that this makes sense, and also whether to update this in v1alpha1, v1alpha2, or both.

**Comment by robscott** [comment]
Thanks for raising this issue! I know we'd previously said that we'd freeze v1alpha1 and instead focus our development on v1alpha2, but maybe if there are enough bugs we should work on a 0.3.1 with a new release-0.3 branch to support a patch here. 

I'll bring this up at our community meeting tomorrow. I think this is a bug, I'm just not sure how broadly we want to apply the patch.

**Comment by robscott** [comment]
We discussed this at today's community meeting and decided that it did not meet the bar for a new v1alpha1 release. Given that, I think it makes sense to make this change in v1alpha2 only. As I've been thinking about it more though, I realized that this was only optional in v1alpha1 to ensure backwards compatibility. IIRC, the intent was to make this field required in v1alpha2. I'll open a PR to make that change so the lack of `omitempty` for this field actually makes sense in v1alpha2.
